homeassistant/auth/auth_store.py

```diff
@@ -228,17 +233,15 @@ def async_remove_refresh_token(self, refresh_token: models.RefreshToken) -> None
         """Remove a refresh token."""
         for user in self._users.values():
             if user.refresh_tokens.pop(refresh_token.id, None):
+                self._token_id_to_user_id.pop(refresh_token.id)
                 self._async_schedule_save()
                 break
 
     @callback
     def async_get_refresh_token(self, token_id: str) -> models.RefreshToken | None:
         """Get refresh token by id."""
-        for user in self._users.values():
-            refresh_token = user.refresh_tokens.get(token_id)
-            if refresh_token is not None:
-                return refresh_token
-
+        if user_id := self._token_id_to_user_id.get(token_id):
+            return self._users[user_id].refresh_tokens.get(token_id)
```

Comment:
It would be a bit faster to cache token_id_to_refresh_token, but I wanted to do the user lookup every time even if its one more lookup to ensure if anything ever goes wrong the user value is the source of truth

Reply:
We could store the userid inside the access token by jwt encoding it. Would avoid the lookup.

Reply:
The performance cost of the lookup is tiny compared the previous cost of searching all the users so I wasn't too concerned about the lookup.  Mostly I was trying to communicate that this will fail securely instead of failing open

Sorry, I wasn't clear about this.